src/Test.UnitTests.Sarif/MockFactory.cs

```diff
@@ -16,6 +16,7 @@ public static IFileSystem MakeMockFileSystem(string fileName, string fileText =
             mock.Setup(fs => fs.PathGetFullPath(It.IsAny<string>())).Returns((string path) => path);
             mock.Setup(fs => fs.FileReadAllText(fileName)).Returns(fileText ?? string.Join(Environment.NewLine, fileLines));
             mock.Setup(fs => fs.FileReadAllLines(fileName)).Returns(fileLines ?? fileText.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries));
+            mock.Setup(x => x.PathGetExtension(It.IsAny<string>())).Returns((string path) => SarifUtilities.PathGetExtension(path));
```

Comment:
>PathGetExtension [](http://example.com/codeflow?start=30&length=16)

this mocking code is a little troubling in that it sets some default behaviors (input agnostic) and others that are exclusive to the file name. 

can we at least separate these setups cleanly in the code?